### PR TITLE
Deployment

### PR DESCRIPTION
**Pull request by produdez**

### CODE REVIEW

File: deploy.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
+docker_command="docker-compose --env-file .env.production"
```
Comment (phhai-hcmut) [inline deploy.sh:1]: This file is not for deploying, it's more like testing production build on local machine

File: app/main.py [hunk 2]
```diff
@@ -60,6 +60,10 @@ def disconnect_db():
 
 if __name__ == "__main__":
     import uvicorn  # type: ignore
+    import json
+
+    print('Environment:')
+    print(json.dumps(settings.dict(), indent=4, sort_keys=True))
```
Comment (phhai-hcmut) [inline app/main.py:66]: Debugging leftover?

Comment (produdez) [reply app/main.py:66]: Ehh i think logging that is kinda helpful :v (intentional)

Comment (phhai-hcmut) [reply app/main.py:66]: This will leak secrets (`CONSUMER_KEY`, ...)

File: deploy.sh (hunk 1 above)
Comment (produdez) [reply deploy.sh:1]: why is that? im not familiar with dev-ops, please explain 😸

Comment (phhai-hcmut) [reply deploy.sh:1]: You deploy to a server, not local machine

Comment (phhai-hcmut) [reply deploy.sh:1]: This script just start docker-compose on local machine, right?

File: app/main.py (hunk 2 above)
Comment (produdez) [reply app/main.py:66]: I dont think logging inside a container counts as leaking right?

Comment (phhai-hcmut) [reply app/main.py:66]: Why not?

File: deploy.sh (hunk 1 above)
Comment (produdez) [reply deploy.sh:1]: yea and if it's on VPS it can also do it right :v

File: app/main.py (hunk 2 above)
Comment (produdez) [reply app/main.py:66]: cause if anyone have access to the container they would already have the .env file? how could some logs compare to that?

File: deploy.sh (hunk 1 above)
Comment (phhai-hcmut) [reply deploy.sh:1]: How are you gonna run it on VPS?

File: app/main.py (hunk 2 above)
Comment (phhai-hcmut) [reply app/main.py:66]: But there is no `.env` file in container?

File: deploy.sh (hunk 1 above)
Comment (produdez) [reply deploy.sh:1]: i was thinking of 
1. `git pull production` (or `git clone` if first time)
2. `setup .env.prod` (skip unless first time)
3. `docker-compose`
But nothing concrete for now, havent got much experience.

Comment (phhai-hcmut) [reply deploy.sh:1]: I'm asking how do you run it on VPS, not what you do

File: app/main.py (hunk 2 above)
Comment (produdez) [reply app/main.py:66]: yea no, so how could it even leak without access to the host?

File: deploy.sh (hunk 1 above)
Comment (phhai-hcmut) [reply deploy.sh:1]: Also why do you put docker-compose command into a variable?

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      source: ./app
+      target: /backend/app
+    command: python -m app.main
+    depends_on: 
+      - database
+    networks:
+      - db_network
+    environment:
+      # deployment related stuffs
+      PYTHON_ENV: 'production'
```
Comment (phhai-hcmut) [inline docker-compose.yml:23]: You can also use `docker-compose` in development

File: deploy.sh (hunk 1 above)
Comment (produdez) [reply deploy.sh:1]: cause typing the whole thing is tedious and uhh i dont want to forget the argument :v, this can be deleted after we finish deploy :3.

Comment (phhai-hcmut) [reply deploy.sh:1]: Why not just write a function then?

Comment (produdez) [reply deploy.sh:1]: what function?

**Review by phhai-hcmut — APPROVED** [review]

**Pull request merged**